src/libsync/abstractcorejob.h

```diff
@@ -122,6 +123,20 @@ class OWNCLOUDSYNC_EXPORT AbstractCoreJobFactory : public QObject
      */
     static void setJobError(CoreJob *job, const QString &errorMessage, const QNetworkReply::NetworkError networkError);
 
+    /**
+     * Factory to create QNetworkRequests with properly set timeout.
+     */
+    template <typename... Params>
+    static OWNCLOUDSYNC_EXPORT QNetworkRequest makeRequest(Params... params)
```

Comment:
hmmm I'm tempted to suggest subclassing and enforcing the use of the new class in the factory but...

Reply:
As long as one just needs a "configured" object, subclassing should not be used in my opinion. Sure, this is not ideal, but the long term goal is to simplify the job infrastructure and eliminate duplicate structures anyway.